library/core/src/main/kotlin/com/freeletics/coredux/ReduxStore.kt

```diff
@@ -148,16 +153,18 @@ private class CoreduxStore<S: Any, A: Any>(
 
 
     override fun subscribe(subscriber: StateReceiver<S>) {
+        if (reducerCoroutine.isCompleted) throw IllegalStateException("CoroutineScope is cancelled")
+
         val receiversIsEmpty = stateReceiversList.isEmpty()
-        stateReceiversList = stateReceiversList + subscriber
+        stateReceiversList += subscriber
```

Comment:
I think StateRecevier List must be thread safe as it can be accessed and changed from reducer "main" coroutine and from any other thread outside of the store, right? Worth adding some locking / synchronization mechanism?